SSH transport may not be respecting reopen interval

It was noticed on @borellim 's system that SSH transport requests seem to be actioned more frequently than the safe open interval.  This is the precise thing that the transport queue is there to limit and therefore this issue requires further investigation.

Here is an excerpt from the daemon log in debug mode. Things to look for:
`SSHException: Error reading SSH protocol banner`, and
`NameError: global name 'clean_remote' is not defined`.

```
08/06/2018 03:16:13 PM <28734> aiida.daemon.runner: [INFO] Received signal to shut down the daemon runner
08/06/2018 03:16:13 PM <28734> aiida.daemon.runner: [INFO] Daemon runner stopped
08/06/2018 03:35:00 PM <29526> aiida.daemon.runner: [INFO] Starting a daemon runner
08/06/2018 03:35:00 PM <29526> aiida.daemon.runner: [DEBUG] Ticking the legacy workflows
08/06/2018 03:35:00 PM <29526> aiida.daemon.runner: [DEBUG] Checking for workflows to manage
08/06/2018 03:35:00 PM <29526> aiida.daemon.runner: [DEBUG] Running execute_steps
08/06/2018 03:35:00 PM <29526> aiida.workflowmanager: [DEBUG] Querying the worflow DB
08/06/2018 03:35:00 PM <29526> aiida.orm.implementation.general.calculation.work.WorkCalculation: [INFO] Loaded process<25108> from saved state
08/06/2018 03:35:00 PM <29526> aiida.aiida_quantumespresso.calculations.pw.PwCalculation: [INFO] Loaded process<25118> from saved state
08/06/2018 03:35:00 PM <29526> aiida.aiida_quantumespresso.calculations.pw.PwCalculation: [INFO] Waiting for calculation<25118> on submit
08/06/2018 03:35:05 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:07 PM <29526> aiida.work.job_processes: [INFO] submitting calculation<25118>
08/06/2018 03:35:07 PM <29526> aiida.scheduler.slurm: [DEBUG] _get_run_line output: ["'srun' '-n' '6' '--hint=nomultithread' '--unbuffered' '/users/mborelli/SIRIUS_builds/20180525/bin/pw.x' '-nk' '6' '-sirius' '-sirius_cfg' '/users/mborelli/sirius_cfg/sirius_config_anton_20180329.json' '-in' 'aiida.in'  > 'aiida.out' "]
08/06/2018 03:35:07 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && whoami
08/06/2018 03:35:09 PM <29526> aiida.execmanager: [DEBUG] [submission of calculation 25118] copying file/folder _aiidasubmit.sh...
08/06/2018 03:35:09 PM <29526> aiida.execmanager: [DEBUG] [submission of calculation 25118] copying file/folder aiida.in...
08/06/2018 03:35:09 PM <29526> aiida.execmanager: [DEBUG] [submission of calculation 25118] copying file/folder .aiida...
08/06/2018 03:35:09 PM <29526> aiida.execmanager: [DEBUG] [submission of calculation 25118] copying file/folder out...
08/06/2018 03:35:09 PM <29526> aiida.execmanager: [DEBUG] [submission of calculation 25118] copying file/folder pseudo...
08/06/2018 03:35:09 PM <29526> aiida.execmanager: [DEBUG] [submission of calculation 25118] copying local file/folder to ./pseudo/Hf-sp.oncvpsp.upf
08/06/2018 03:35:09 PM <29526> aiida.execmanager: [DEBUG] [submission of calculation 25118] copying local file/folder to ./pseudo/Ru_ONCV_PBE-1.0.upf
08/06/2018 03:35:09 PM <29526> aiida.scheduler.slurm: [INFO] submitting with: sbatch '_aiidasubmit.sh'
08/06/2018 03:35:09 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/scratch/snx3000/mborelli/aiida_run/0a/e9/e72c-7d59-42fa-9dc5-dccb5e5f51e0' && sbatch '_aiidasubmit.sh'
08/06/2018 03:35:11 PM <29526> aiida.work.job_processes: [INFO] submitting calculation<25118> successful
08/06/2018 03:35:11 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:11 PM <29526> aiida.work.persistence: [DEBUG] Persisting process<25118>
08/06/2018 03:35:11 PM <29526> aiida.aiida_quantumespresso.calculations.pw.PwCalculation: [INFO] Waiting for calculation<25118> on update_scheduler
08/06/2018 03:35:16 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:17 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:35:17 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:17 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:19 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:35:19 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:35:19 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:24 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:25 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:35:25 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:25 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:26 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:35:26 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:35:26 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:30 PM <29526> aiida.daemon.runner: [DEBUG] Ticking the legacy workflows
08/06/2018 03:35:30 PM <29526> aiida.daemon.runner: [DEBUG] Checking for workflows to manage
08/06/2018 03:35:30 PM <29526> aiida.daemon.runner: [DEBUG] Running execute_steps
08/06/2018 03:35:30 PM <29526> aiida.workflowmanager: [DEBUG] Querying the worflow DB
08/06/2018 03:35:31 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:33 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:35:33 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:33 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:33 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:35:33 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:35:33 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:39 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:40 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:35:40 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:40 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:41 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:35:41 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:35:41 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:46 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:47 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:35:47 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:47 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:48 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:35:48 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:35:48 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:53 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:35:55 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:35:55 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:55 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:35:55 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:35:55 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:35:55 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:00 PM <29526> aiida.daemon.runner: [DEBUG] Ticking the legacy workflows
08/06/2018 03:36:00 PM <29526> aiida.daemon.runner: [DEBUG] Checking for workflows to manage
08/06/2018 03:36:00 PM <29526> aiida.daemon.runner: [DEBUG] Running execute_steps
08/06/2018 03:36:00 PM <29526> aiida.workflowmanager: [DEBUG] Querying the worflow DB
08/06/2018 03:36:01 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:02 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:36:02 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:02 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:03 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:36:03 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:36:03 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:08 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:09 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:36:09 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:09 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:10 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:36:10 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:36:10 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:15 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:16 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:36:16 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:16 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:17 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:36:17 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:36:17 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:22 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:24 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:36:24 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:24 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:25 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:36:25 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:36:25 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:30 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:31 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:36:31 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:31 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:46 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:36:46 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:36:46 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:36:46 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:46 PM <29526> aiida.daemon.runner: [DEBUG] Ticking the legacy workflows
08/06/2018 03:36:46 PM <29526> aiida.daemon.runner: [DEBUG] Checking for workflows to manage
08/06/2018 03:36:46 PM <29526> aiida.daemon.runner: [DEBUG] Running execute_steps
08/06/2018 03:36:46 PM <29526> aiida.workflowmanager: [DEBUG] Querying the worflow DB
08/06/2018 03:36:51 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:53 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118>
08/06/2018 03:36:53 PM <29526> aiida.scheduler.slurm: [DEBUG] squeue command: SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:53 PM <29526> aiida.transport.SshTransport: [DEBUG] Command to be executed: cd '/users/mborelli' && SLURM_TIME_FORMAT='standard' squeue --noheader -o '%i^^^%t^^^%r^^^%B^^^%u^^^%D^^^%C^^^%R^^^%P^^^%l^^^%M^^^%S^^^%j^^^%V' --jobs=8928768
08/06/2018 03:36:54 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:36:54 PM <29526> aiida.scheduler: [DEBUG] Datetime to serialize in JobInfo is naive, this should be fixed!
08/06/2018 03:36:54 PM <29526> aiida.work.job_processes: [INFO] updating calculation<25118> successful
08/06/2018 03:36:54 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:36:59 PM <29526> aiida.work.transports: [DEBUG] Transport request opening transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:37:14 PM <29526> paramiko.transport: [ERROR] Exception: Error reading SSH protocol banner
08/06/2018 03:37:14 PM <29526> paramiko.transport: [ERROR] Traceback (most recent call last):
08/06/2018 03:37:14 PM <29526> paramiko.transport: [ERROR]   File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/paramiko/transport.py", line 1893, in run
08/06/2018 03:37:14 PM <29526> paramiko.transport: [ERROR]     self._check_banner()
08/06/2018 03:37:14 PM <29526> paramiko.transport: [ERROR]   File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/paramiko/transport.py", line 2049, in _check_banner
08/06/2018 03:37:14 PM <29526> paramiko.transport: [ERROR]     'Error reading SSH protocol banner' + str(e)
08/06/2018 03:37:14 PM <29526> paramiko.transport: [ERROR] SSHException: Error reading SSH protocol banner
08/06/2018 03:37:14 PM <29526> paramiko.transport: [ERROR] 
08/06/2018 03:37:14 PM <29526> aiida.transport.SshTransport: [ERROR] Error connecting through SSH: [SSHException] Error reading SSH protocol banner, connect_args were: {'username': u'mborelli', 'allow_agent': True, 'sock': <aiida.transport.util._DetachedProxyCommand object at 0x7fb5cfdd1390>, 'gss_deleg_creds': False, 'key_filename': u'/home/mborelli/.ssh/id_rsa-cscs2', 'compress': True, 'look_for_keys': True, 'timeout': 60, 'gss_kex': False, 'gss_host': u'daint.cscs.ch', 'port': 22, 'gss_auth': False}
08/06/2018 03:37:14 PM <29526> aiida.work.transports: [WARNING] exception occurred while trying to open the transport:
Error reading SSH protocol banner
08/06/2018 03:37:14 PM <29526> aiida.work.transports: [ERROR] Exception whilst using transport:
Traceback (most recent call last):
  File "/home/mborelli/repos/aiida_dev/aiida-core/aiida/work/transports.py", line 100, in request_transport
    yield transport_request.future
  File "/home/mborelli/repos/aiida_dev/aiida-core/aiida/work/job_processes.py", line 115, in task_update_job
    transport = yield request
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 3, in raise_exc_info
SSHException: Error reading SSH protocol banner

08/06/2018 03:37:14 PM <29526> aiida.work.transports: [DEBUG] Transport request closing transport for AuthInfo for marco.borelli@epfl.ch on daint5
08/06/2018 03:37:15 PM <29526> aiida.aiida_quantumespresso.calculations.pw.PwCalculation: [REPORT] [25118|JobProcess_aiida_quantumespresso.calculations.pw:PwCalculation|on_except]: Traceback (most recent call last):
  File "/home/mborelli/repos/aiida_dev/aiida-core/aiida/work/job_processes.py", line 271, in execute
    job_done = yield task_update_job(calc, transport_queue, self._cancel_flag)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/home/mborelli/repos/aiida_dev/aiida-core/aiida/work/job_processes.py", line 115, in task_update_job
    transport = yield request
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 3, in raise_exc_info
SSHException: Error reading SSH protocol banner

08/06/2018 03:37:15 PM <29526> tornado.application: [ERROR] Future exception was never retrieved: Traceback (most recent call last):
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/home/mborelli/repos/aiida_dev/kiwipy/kiwipy/rmq/tasks.py", line 141, in _on_task
    result = yield task_future
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/mborelli/repos/aiida_dev/kiwipy/kiwipy/futures.py", line 43, in result
    return super(Future, self).result(timeout=0.)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/home/mborelli/repos/aiida_dev/aiida-core/aiida/work/job_processes.py", line 271, in execute
    job_done = yield task_update_job(calc, transport_queue, self._cancel_flag)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/home/mborelli/repos/aiida_dev/aiida-core/aiida/work/job_processes.py", line 115, in task_update_job
    transport = yield request
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 3, in raise_exc_info
SSHException: Error reading SSH protocol banner
08/06/2018 03:37:15 PM <29526> aiida.work.persistence: [DEBUG] Persisting process<25108>
08/06/2018 03:37:15 PM <29526> aiida.orm.implementation.general.calculation.work.WorkCalculation: [REPORT] [25108|PwBaseWorkChain|_handle_unexpected_failure]: Node 25118 does not have an output with link output_parameters
08/06/2018 03:37:15 PM <29526> aiida.orm.implementation.general.calculation.work.WorkCalculation: [REPORT] [25108|PwBaseWorkChain|_handle_unexpected_failure]: failure of PwCalculation<25118> could not be handled for the second consecutive time
08/06/2018 03:37:17 PM <29526> tornado.application: [ERROR] Exception in callback <functools.partial object at 0x7fb5cfc3ac58>
Traceback (most recent call last):
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/ioloop.py", line 605, in _run_callback
    ret = callback()
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/stack_context.py", line 277, in null_wrapper
    return fn(*args, **kwargs)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/ioloop.py", line 626, in _discard_future_result
    future.result()
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/plumpy/processes.py", line 983, in step_until_terminated
    yield self.step()
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/tornado/gen.py", line 1069, in run
    yielded = self.gen.send(value)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/plumpy/processes.py", line 972, in step
    self.transition_to(next_state)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/plumpy/base/state_machine.py", line 326, in transition_to
    *sys.exc_info()[1:])
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/plumpy/base/state_machine.py", line 339, in transition_failed
    raise_(type(exception), exception, trace)
  File "/home/mborelli/venvs/aiida_dev/lib/python2.7/site-packages/plumpy/base/state_machine.py", line 319, in transition_to
    self.on_terminated()
  File "/home/mborelli/repos/aiida_dev/aiida-quantumespresso/aiida_quantumespresso/common/workchain/base/restart.py", line 229, in on_terminated
    called_descendant.out.remote_folder._clean()
  File "/home/mborelli/repos/aiida_dev/aiida-core/aiida/orm/data/remote.py", line 169, in _clean
    clean_remote(transport, remote_dir)
NameError: global name 'clean_remote' is not defined
08/06/2018 03:37:17 PM <29526> aiida.daemon.runner: [DEBUG] Ticking the legacy workflows
08/06/2018 03:37:17 PM <29526> aiida.daemon.runner: [DEBUG] Checking for workflows to manage
08/06/2018 03:37:17 PM <29526> aiida.daemon.runner: [DEBUG] Running execute_steps
08/06/2018 03:37:17 PM <29526> aiida.workflowmanager: [DEBUG] Querying the worflow DB
08/06/2018 03:37:47 PM <29526> aiida.daemon.runner: [DEBUG] Ticking the legacy workflows
08/06/2018 03:37:47 PM <29526> aiida.daemon.runner: [DEBUG] Checking for workflows to manage
08/06/2018 03:37:47 PM <29526> aiida.daemon.runner: [DEBUG] Running execute_steps
08/06/2018 03:37:47 PM <29526> aiida.workflowmanager: [DEBUG] Querying the worflow DB
^C
```

--- a/aiida/backends/tests/work/test_transport.py
+++ b/aiida/backends/tests/work/test_transport.py
@@ -99,3 +99,35 @@ class TestTransportQueue(AiidaTestCase):
                 loop.run_sync(lambda: test())
         finally:
             self.authinfo.get_transport().__class__.open = original
+
+    def test_safe_interval(self):
+        """Verify that the safe interval for a given in transport is respected by the transport queue."""
+
+        # Temporarily set the safe open interval for the default transport to a finite value
+        transport_class = self.authinfo.get_transport().__class__
+        original_interval = transport_class._DEFAULT_SAFE_OPEN_INTERVAL
+
+        try:
+            transport_class._DEFAULT_SAFE_OPEN_INTERVAL = 0.25
+
+            import time
+            queue = TransportQueue()
+            loop = queue.loop()
+
+            time_start = time.time()
+
+            @coroutine
+            def test(iteration):
+                trans = None
+                with queue.request_transport(self.authinfo) as request:
+                    trans = yield request
+                    time_current = time.time()
+                    time_elapsed = time_current - time_start
+                    time_minimum = trans.get_safe_open_interval() * (iteration + 1)
+                    self.assertTrue(time_elapsed > time_minimum, 'transport safe interval was violated')
+
+            for i in range(5):
+                loop.run_sync(lambda: test(i))
+
+        finally:
+            transport_class._DEFAULT_SAFE_OPEN_INTERVAL = original_interval

